### PR TITLE
Use cacheable routes for course elements and element extensions

### PR DESCRIPTION
**Pull request by nwalters512**
This takes us a step closer to removing these non-cached routes:

https://github.com/PrairieLearn/PrairieLearn/blob/51d26b76795eedb28c510f8770dc5df52d76c7e3/apps/prairielearn/src/server.js#L842-L865

Note that these routes are still used in local dev where `course.commit_hash` is not populated. Probably what we'll end up doing in the long run is mounting these routes only in dev mode. We really want everyone in production to only use cacheable routes.

### CODE REVIEW

File: apps/prairielearn/src/question-servers/freeform.js [hunk 1]
```diff
@@ -1165,6 +1161,12 @@ async function renderPanel(panel, codeCaller, variant, submission, course, local
   data.options.client_files_question_url = locals.clientFilesQuestionUrl;
   data.options.client_files_course_url = locals.clientFilesCourseUrl;
   data.options.client_files_question_dynamic_url = locals.clientFilesQuestionGeneratedFileUrl;
+  data.options.course_element_files_url = assets
+    .courseElementAssetPath(course.commit_hash, locals.urlPrefix, '')
+    .replace(/\/+$/, '');
+  data.options.course_element_extension_files_url = assets
+    .courseElementExtensionAssetPath(course.commit_hash, locals.urlPrefix, '')
+    .replace(/\/+$/, '');
```
Comment (mwest1066) [inline apps/prairielearn/src/question-servers/freeform.js:1169]: This `replace()` stuff feels a bit nasty. Could we instead have `courseElementAssetPath()` accept a `null` (or `undefined`?) third argument and in that case return the path without the trailing slash?

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/question-servers/freeform.js (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/question-servers/freeform.js]: Very fair point! How does 1ca4ad3 look?

File: apps/prairielearn/src/question-servers/freeform.js [hunk 2]
```diff
@@ -296,24 +296,20 @@ function getElementClientFiles(data, elementName, context) {
...
-    dataCopy.options.client_files_element_url = path.posix.join(
+    dataCopy.options.client_files_element_url = assets.courseElementAssetPath(
+      context.course.commit_hash,
       data.options.base_url,
-      'elements',
-      elementName,
-      'clientFilesElement',
+      `${elementName}/clientFilesElement`,
     );
+    console.log(dataCopy.options.client_files_element_url);
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/question-servers/freeform.js]: Removed in a00efcf.

File: apps/prairielearn/src/question-servers/freeform.js (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/src/question-servers/freeform.js]: Much cleaner! 🙏

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**